core/src/Kokkos_MathematicalFunctions.hpp

```diff
@@ -227,6 +227,25 @@ KOKKOS_IMPL_MATH_UNARY_PREDICATE(isnan)
 #undef KOKKOS_IMPL_MATH_UNARY_PREDICATE
 #undef KOKKOS_IMPL_MATH_BINARY_FUNCTION
 #undef KOKKOS_IMPL_MATH_NAN
+
+// clang-format off
+#ifdef KOKKOS_ENABLE_SYCL
+KOKKOS_INLINE_FUNCTION int abs(int n) { return sycl::abs(n); }
+KOKKOS_INLINE_FUNCTION long abs(long n) { return sycl::abs(n); }
+KOKKOS_INLINE_FUNCTION long long abs(long long n) { return sycl::abs(n); }
+KOKKOS_INLINE_FUNCTION float abs(float x) { return sycl::fabs(x); }
+KOKKOS_INLINE_FUNCTION double abs(double x) { return sycl::fabs(x); }
+#else
+KOKKOS_INLINE_FUNCTION int abs(int n) { using std::abs;  return abs(n); }
+KOKKOS_INLINE_FUNCTION long abs(long n) { using std::labs;  return labs(n); }
+KOKKOS_INLINE_FUNCTION long long abs(long long n) { using std::llabs;  return llabs(n); }
```

Comment:
While I don't think it matters in this case, there are overloads for `std::abs` for `long`, `long long`, `float`, `double` and `long double`.

Reply:
I am trying to find something that works with the CUDA math functions and these overloads are missing unfortunately

Reply:
CUDA provides the following (unqualified) functions
```C++
__device__ ​ int abs ( int  a );
__device__ ​ long int labs ( long int  a );
__device__ ​ long long int 	llabs ( long long int a );
__device__ ​ float fabsf ( float  x );
__device__ ​ double fabs ( double  x );

Reply:
Got it.  Thanks for the explanation.

Reply:
I was wrong.  I based my answer on the documented function in the [CUDA math API](https://docs.nvidia.com/cuda/cuda-math-api/modules.html#modules) but I checked  the hardware counters with `nvprof` and `::abs((int)i)` does not give any floating-point instruction.